spec/controllers/merchants/revenue_controller_spec.rb

```diff
@@ -44,70 +44,80 @@
   end
 
   describe "GET /:id/revenue" do
-    xit "returns the total revenue for that merchant across all transactions" do
-      m = FactoryGirl.create(:merchant, id: 1)
-      item = FactoryGirl.create(:item, merchant: m, unit_price: 100)
-      c = FactoryGirl.create(:customer, id: 1)
-      invoice_1 = FactoryGirl.create(:invoice, customer: c, merchant: m)
-      invoice_2 = FactoryGirl.create(:invoice, customer: c, merchant: m)
-      FactoryGirl.create(:invoice_item,
+    it "returns the total revenue for that merchant across all transactions" do
+      merchant = FactoryGirl.create(:merchant, id: 1)
+
+      item = FactoryGirl.create(:item, merchant: merchant, unit_price: 100.00)
+
+      invoice_1 = FactoryGirl.create(:invoice, merchant: merchant)
+      invoice_2 = FactoryGirl.create(:invoice, merchant: merchant)
+
+      FactoryGirl.create(
+        :invoice_item,
         invoice: invoice_1,
         item: item,
-        quantity: 2,
-        unit_price: 100.00
+        quantity: 3,
+        unit_price: item.unit_price
       )
-      FactoryGirl.create(:invoice_item,
+      FactoryGirl.create(
+        :invoice_item,
         invoice: invoice_2,
         item: item,
-        quantity: 3,
-        unit_price: 100.00
+        quantity: 4,
+        unit_price: item.unit_price
       )
 
+      FactoryGirl.create(:transaction, invoice: invoice_1, result: "success")
+      FactoryGirl.create(:transaction, invoice: invoice_2, result: "success")
```

Comment:
**This (creating the transactions) was actually the only part that was necessary. The rest is just refactoring that happened while I tried to figure out the issue.**